File: app/src/main/java/com/bypassmobile/octo/rest/GithubEndpoint.java
package com.bypassmobile.octo.rest;


import com.bypassmobile.octo.model.User;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface GithubEndpoint {

    @GET("/users/{id}")
    public Call<List<User>> getUser(@Path("id") String user);

    @GET("/users/{id}/following")
    public Call<List<User>> getFollowingUser(@Path("id") String user);

    @GET("/orgs/{id}/members")
    public Call<List<User>> getOrganizationMember(@Path("id") String organization);
}

File: app/src/main/java/com/bypassmobile/octo/rest/GithubEndpoint.kt
package com.bypassmobile.octo.rest

import com.bypassmobile.octo.model.User
import io.reactivex.Single

import retrofit2.http.GET
import retrofit2.http.Path

interface GithubEndpoint {

    @GET("/users/{id}")
    fun getUser(@Path("id") user: String): Single<User>

    @GET("/users/{id}/following")
    fun getFollowingUser(@Path("id") user: String): Single<List<User>>

    @GET("/orgs/{id}/members")
    fun getOrganizationMember(@Path("id") organization: String): Single<List<User>>

    companion object {
        val SERVER = "https://api.github.com"
    }
}
